pkg/mutation/mutators/modifyset/modify_set_mutator.go

```diff
@@ -138,7 +138,7 @@ func MutatorForModifySet(modifySet *mutationsunversioned.ModifySet) (*Mutator, e
 		return nil, fmt.Errorf("modifyset %s can't change metadata", modifySet.GetName())
 	}
 
-	if path.Nodes[len(path.Nodes)-1].Type() == parser.ListNode {
+	if len(path.Nodes) > 0 && path.Nodes[len(path.Nodes)-1].Type() == parser.ListNode {
```

Comment:
empty spec leads to empty path which panics. I've added unit tests for all mutators that don't have an empty spec test case to enforce graceful handling of an empty spec.